iele-gas.md

```diff
@@ -720,47 +720,57 @@ Note: These are all functions as the operator `#compute` has already loaded all
  // ----------------------------------------------------------
     rule G*(GAVAIL, GLIMIT, REFUND) => GAVAIL +Int minInt((GLIMIT -Int GAVAIL)/Int 2, REFUND)
 
-    syntax Int ::= Cdiv     ( Schedule , Int , Int )       [function]
-                 | Cmul     ( Schedule , Int , Int )       [function]
-                 | Cexp     ( Schedule , Int , Int )       [function]
-                 | Cexpmod  ( Schedule , Int , Int , Int ) [function]
- // -----------------------------------------------------------------
+    syntax Int ::= Cmul     ( Schedule , Int , Int )             [function]
+                 | Ckara    ( Int , Int )                        [function]
+                 | Cdiv     ( Schedule , Int , Int )             [function]
+                 | Cexp     ( Schedule , Int , Int , Int )       [function]
+                 | Cexpmod  ( Schedule , Int , Int , Int , Int ) [function]
+ // -----------------------------------------------------------------------
     rule Cmul(SCHED, L1, L2) => Cmul(SCHED, L2, L1)
       requires L2 >Int L1
+
     rule Cmul(SCHED, L1, L2) =>
-        Gmulkara < SCHED > *Int L1 *Int #overApproxKara(L2) /Int L2 +Int
-        Gmulword < SCHED > *Int L1 +Int
+        Gmulkara < SCHED > *Int Ckara(L1, L2) +Int
+        Gmulword < SCHED > *Int (L1 +Int L2) +Int
         Gmul < SCHED >
-      requires L1 >Int L2
+      [owise]
       // Note that if L2 is low enough (< 32) then #overApproxKara(L2) = L2 * L2
 
-    rule Cexp(SCHED, L1, W2) =>
-        Gexpkara < SCHED > *Int #overApproxKara(L1 *Int W2) +Int
-        Gexpword < SCHED > *Int L1 +Int
-        Gexp < SCHED >
+    rule Ckara(L1, L2) => L1 *Int #overApproxKara(L2) /Int L2
+      requires L1 >=Int L2
 
-    rule Cexpmod(SCHED, LB, LEX, LM) =>
-        Cdiv(SCHED, LB, LM) +Int
-        Gexpmodmul < SCHED > *Int LEX *Int Cmul(SCHED, LM, LM) +Int
-        Cdiv(SCHED, 2 *Int LM, LM) +Int
-        Gexpmodwordm < SCHED > *Int LM +Int
-        Gexpmodworde < SCHED > *Int LEX +Int
-        Gexpmod < SCHED >
+    rule Ckara(L1, L2) => L2 *Int #overApproxKara(L1) /Int L1
```

Comment:
should this have `[owise]` or a `requires` clause?

Reply:
what's wrong with owise?

Reply:
It would be fine, but this rule doesn't have `owise`, does it?